docs(ng-schematics): add Protractor migration info (#10741)
puppeteer/puppeteer: docs/ng-schematics.md

```bash
npm run test
```

## Migrating from Protractor

### Browser

> Puppeteer has it's own [`browser`](https://pptr.dev/api/puppeteer.browser) that exposes different API compared the Protractor one.

Reviewer: ```suggestion
Puppeteer has its own [`browser`](https://pptr.dev/api/puppeteer.browser) that exposes different API compared to the one exposed by Protractor.
```

In most scenarios you will create a new page for each of your test:

```ts
import puppeteer from 'puppeteer';

(async () => {
  const browser = await puppeteer.launch();

  it('should work', () => {
    const page = await browser.newPage();

    // Query elements
    const element = await page.$('my-component');

    // Do actions
    await element.click();
  });

  await browser.close();
})();
```

### Query Selectors

> Puppeteer support different type of querySelectors - CSS, Aria, Text, XPath, Pierce.

Reviewer: ```suggestion
Puppeteer supports multiple types of selectors, namely, the CSS, ARIA, text, XPath and pierce selectors.
```

> Bellow is a table representing a map between [Protractor By](https://www.protractortest.org/#/api?view=ProtractorBy) and Puppeteer querySelectors.

Reviewer: ```suggestion
The following table shows Puppeteer's equivalents to [Protractor By](https://www.protractortest.org/#/api?view=ProtractorBy).
```

> > For better reliable and less flakiness you can try our

Reviewer: ```suggestion
> For improved reliability and reduced flakiness try our
```

> **Experimental** [Locators API](https://pptr.dev/guides/locators)

| By                | Protractor code                                                   | Puppeteer querySelector                                      |
| ----------------- | ----------------------------------------------------------------- | ------------------------------------------------------------ |
| CSS               | Single: `$(by.css('<CSS>'))` <br> Multiple: `$$(by.css('<CSS>'))` | Single: `page.$('<CSS>')` <br> Multiple: `page.$$('<CSS>')`  |
| Id                | `$(by.id('<ID>'))`                                                | `page.$('#<ID>')`                                            |
| CssContainingText | `$(by.cssContainingText('<CSS>', '<TEXT>'))`                      | `page.$('<CSS> ::-p-text(<TEXT>)')` `                        |

> | DeepCss           | `$(by.deepCss('<CSS>'))`                                          | `page.$('<CSS>')`                                            |

Reviewer: ```suggestion
| DeepCss           | `$(by.deepCss('<CSS>'))`                                          | `page.$(':scope >>> <CSS>')`                                            |
```

| XPath             | `$(by.xpath('<XPATH>'))`                                          | `page.$('::-p-xpath(<XPATH>)')`                              |
| JS                | `$(by.js('document.querySelector("<CSS>")'))`                     | `page.evaluateHandle(() => document.querySelector('<CSS>'))` |

> > For advance usage such as Protractor's `by.addLocator` you can check Puppeteer's [Custom selectors](https://pptr.dev/guides/query-selectors#custom-selectors).

Reviewer: ```suggestion
> For advanced use cases such as Protractor's `by.addLocator` you can check Puppeteer's [Custom selectors](https://pptr.dev/guides/query-selectors#custom-selectors).
```
